core/templates/pages/topics-and-skills-dashboard-page/modals/delete-skill-modal.component.spec.ts

```diff
@@ -92,4 +110,40 @@ describe('Assign Skill to Topic Modal Component', () => {
     expect(componentInstance.topicsAssignments).toEqual(testSkills);
     expect(componentInstance.topicsAssignmentsAreFetched).toBeTrue();
   });
+
+  it('should allow skill deletion', fakeAsync(() => {
+    let topicsAndSkillsDashboardBackendApiService = TestBed.inject(
+      TopicsAndSkillsDashboardBackendApiService);
+    componentInstance.skillId = 'skill_id';
+    componentInstance.topicsAssignmentsAreFetched = false;
+    spyOn(
+      topicsAndSkillsDashboardBackendApiService,
+      'fetchTopicIdToDiagnosticTestSkillIdsAsync'
+    ).and.returnValue(Promise.resolve({
+      topicIdToDiagnosticTestSkillIds: {topic_id: []}
+    }));
+    componentInstance.fetchTopicAssignmentsForSkill();
+    tick(50);
+    expect(componentInstance.allowDeletionOfSkill).toBeTrue();
+  }));
+
+  it('should not allow skill deletion', fakeAsync(() => {
+    let topicsAndSkillsDashboardBackendApiService = TestBed.inject(
+      TopicsAndSkillsDashboardBackendApiService);
+    componentInstance.skillId = 'skill_id';
+    componentInstance.topicsAssignmentsAreFetched = false;
+    spyOn(
+      topicsAndSkillsDashboardBackendApiService,
+      'fetchTopicIdToDiagnosticTestSkillIdsAsync'
+    ).and.returnValue(Promise.resolve({
+      topicIdToDiagnosticTestSkillIds: {
+        test_id: ['skill_id'],
```

Comment:
It's slightly confusing that the keys for the data are named topic_id in one place and test_id in another. Could you please clarify this?

Reply:
updated